src/CodeGenerator/GeneratorTrait.php

```diff
@@ -44,6 +49,8 @@ protected function ensureDirectory(string $dir)
      *
      * @throws LogicException
      * @throws GenerateCodeException
+     * @return void
+     * @psalm-assert non-empty-string $this->outputDirectory
```

Comment:
Not sure this works with private state :thinking:

Reply:
Yes it does

Reply:
Here you go: https://psalm.dev/r/205038dd2c